app/main.py

```diff
@@ -1 +1,25 @@
-# write your code here
+from __future__ import annotations
+
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    vaccine_error_msg = []
+    mask_error_msg = []
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError as e:
+            print(f"Vaccine error for {friend['name']}: {e}")
+            vaccine_error_msg.append(e)
+        except NotWearingMaskError as e:
+            print(e)
+            mask_error_msg.append(e)
+
+    if not vaccine_error_msg and not mask_error_msg:
+        return f"Friends can go to {cafe.name}"
+    if vaccine_error_msg:
+        return "All friends should be vaccinated"
+    if not vaccine_error_msg and mask_error_msg:
+        return f"Friends should buy {len(mask_error_msg)} masks"
```

Comment:
The condition `if not vaccine_error_msg and mask_error_msg:` is redundant because this case is already covered by the previous checks. You can simplify the logic by removing this condition and directly returning the mask-related message if `mask_error_msg` is not empty.